components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -0,0 +1,43 @@
+package handlers
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
+	"github.com/nats-io/nats.go"
+	"github.com/pkg/errors"
+)
+
+// compile time check
+var _ NatsInterface = &Nats{}
+
+type NatsInterface interface {
```

Comment:
The initial role of the "handlers" package was to extract, if possible the logic of initialization/create/delete subscription from the reconcile code and put it here. Perhaps we will expose such a back-end messaging handler as an "add-on" extension to kyma-eventing.
This is why for "beb" we have defined:
```
type Interface interface {
	Initialize(cfg env.Config)
	SyncBebSubscription(subscription *eventingv1alpha1.Subscription, apiRule *apigatewayv1alpha1.APIRule) (bool, error)
	DeleteBebSubscription(subscription *eventingv1alpha1.Subscription) error
}
```
The "nats" handler could also be extended to add the similar functionality in addition to:
```
type NatsInterface interface {
	Initialize(cfg env.NatsConfig) error
}
```
The goal is to isolate the "back-end" specific code from the "controller" code to make easier the supporting of other messaging back-ends.

Reply:
Changed the interface accordingly.